Added docs around IMemberPartialViewCacheInvalidator (#7133)
umbraco/UmbracoDocs: 13/umbraco-cms/reference/cache/imemberpartialviewcacheinvalidator.md

# ICacheRefresher

This section describes what IMemberPartialViewCacheInvalidator is, what it's default implementation does and how to customize it.

> ## What is an IMemberPartialViewCacheInvalidator

Reviewer: ```suggestion
## What is an IMemberPartialViewCacheInvalidator?
```

> This interface is used to isolate the logic that needs to run to invalidate parts of the PartialView cache when a member is updated

Reviewer: ```suggestion
This interface is used to isolate the logic that used invalidate parts of the partial view cache when a member is updated.
```

> ## Why do we need to partialy invalidate the partialView cache

Reviewer: ```suggestion
## Why do we need to partially invalidate the partial view cache?
```

> Since some of the razor templates might show data that is retrieved from a member object and those templates might be cached by using the partial caching mechanism (i.e. `@await Html.CachedPartialAsync("member",Model,TimeSpan.FromDays(1), cacheByMember:true)`), we need to remove those cached partials when a member is updated.

Reviewer: ```suggestion
Razor templates showing data retrieved from a member object can be cached as partial views via:

```
@await Html.CachedPartialAsync("member",Model,TimeSpan.FromDays(1), cacheByMember: true)
```

To ensure we don't display out of date information, we need to remove those cached partials when a member is updated.
```

## Where is it used

> This interface is called from the MemberCacheRefresher which is called every time a member is updated.

Reviewer: ```suggestion
This interface is called from the member cache refresher (`MemberCacheRefresher`) which is invoked every time a member is updated.
```

> ## Details of the implementation

Reviewer: ```suggestion
## Details of the default implementation
```

> When a razor template partial is cached trough `Html.CachedPartialAsync` and `cacheByMember` is set to `true`, the extension method will append the memberId of the currently logged in member and a marker (i.e. `-m1015-`) to the partialView chachekey.

Reviewer: ```suggestion
Razor template partials are cached through a call to `Html.CachedPartialAsync` with `cacheByMember` set to `true`. This will append the id of the currently logged in member with a marker to the partial view cache key.  For example, `-m1015-`. 
```

> When the `ClearPartialViewCacheItems` method is called it will clear all PartialView cacheItems that have the memberId marker for all passed in members.

Reviewer: ```suggestion

When the `ClearPartialViewCacheItems` method is called it will clear all cache items that match the marker for the updated members.
```

> Since it is possible to call the `Html.CachedPartialAsync` with `cacheByMember` set to `true` while there is no member logged in, it will also clear all cache items with an empty member marker (i.e. `-m-`)

Reviewer: ```suggestion

Since it is possible to call the `Html.CachedPartialAsync` with `cacheByMember` set to `true` while there is no member logged in, it will also clear all cache items with an empty member marker (i.e. `-m-`)
```

## Customizing the implementation

> You can replace the default implementation like usual by removing the default and registering your own in a composer.

Reviewer: ```suggestion
You can replace the default implementation by removing it and registering your own in a composer.
```

# Partial view cache refresher for members

This section describes the `IMemberPartialViewCacheInvalidator` interface, what it's default implementation does and how to customize it.

## What is an IMemberPartialViewCacheInvalidator?

This interface is used to isolate the logic that invalidates parts of the partial view cache when a member is updated.

## Why do we need to partially invalidate the partial view cache?

> Razor templates showing data retrieved from a member object can be cached as partial views via:

Reviewer 2: This line is incomplete. Can you update it for both v13 and v16?

## Where is it used?

This interface is called from the member cache refresher (`MemberCacheRefresher`) which is invoked every time a member is updated.

## Details of the default implementation

Razor template partials are cached through a call to `Html.CachedPartialAsync` with `cacheByMember` set to `true`. This will append the id of the currently logged in member with a marker to the partial view cache key.  For example, `-m1015-`. 

When the `ClearPartialViewCacheItems` method is called it will clear all cache items that match the marker for the updated members.

> Since it is possible to call the `Html.CachedPartialAsync` with `cacheByMember` set to `true` while there is no member logged in, it will also clear all cache items with an empty member marker (i.e. `-m-`)

Reviewer 2: ```suggestion
If no member is logged in during caching, items with an empty member marker (for example, `-m-`) are also cleared.
```
